### DISCUSSION TITLE
format datetime to a human form

### DISCUSSION THREAD

**Discussion by W2AlharbiMe** [symfony/symfony#42539] [Q&A] [answered]
Hello, i was wondering is there a function in twig like `diffForHumans` in laravel ?

it should convert the datetime to this format:
```
53 mins ago
2 hours ago
3 days ago
```

**Comment by faizanakram99** [comment, marked as answer]
It's not a laravel function. It's part of carbon PHP package https://carbon.nesbot.com/ .

You should be able to call `diffForHumans()` on any carbon object in PHP as well as twig. If you're using doctrine, register both `datetime` and `datetime_immutable` as carbon objects and all your `datetime` fields would be instances of Carbon.

See the docs I linked above, it has detailed guide about symfony integration.

Also you can create twig extensions for stuff like converting datetime string to carbon objects (if needed)
https://symfony.com/doc/current/templating/twig_extension.html

**Reply by W2AlharbiMe** [reply]
Thank you so much for your help

**Comment by alessandro-podo** [comment]
you can try this package https://github.com/KnpLabs/KnpTimeBundle
i use it, and it works fine

**Reply by W2AlharbiMe** [reply]
Thank you so much for your help
